diff -Nur lids-1.2.2-2.4.28/lidstools-0.5.6/src/Makefile.in lids-1.2.2-2.4.28-greenphone/lidstools-0.5.6/src/Makefile.in
--- lids-1.2.2-2.4.28/lidstools-0.5.6/src/Makefile.in	2004-02-17 16:20:27.000000000 +1000
+++ lids-1.2.2-2.4.28-greenphone/lidstools-0.5.6/src/Makefile.in	2007-02-13 08:38:37.000000000 +1000
@@ -81,7 +81,7 @@
 install_sh = @install_sh@
 sources = read_pw.c rmd160.c sig_rmd160.c time.c kernel_inc.h lids_capflag.h lidstools.h rmd160.h  sigs.h lids_capflag.c lidsif.h
 
-CFLAGS = -I@KERNEL_INC@ -DCONFIG_LIDS
+CFLAGS = -I@KERNEL_INC@ -DCONFIG_LIDS @CFLAGS@
 
 sbin_PROGRAMS = lidsadm lidsconf 
 lidsadm_SOURCES = lidsadm.c $(sources)
