diff -Nur lids-1.2.2-2.4.28/lidstools-0.5.6/configure lids-1.2.2-2.4.28-greenphone/lidstools-0.5.6/configure
--- lids-1.2.2-2.4.28/lidstools-0.5.6/configure	2004-09-15 19:22:38.000000000 +1000
+++ lids-1.2.2-2.4.28-greenphone/lidstools-0.5.6/configure	2007-02-12 15:50:20.000000000 +1000
@@ -2980,9 +2980,9 @@
 if eval "test \"\${$as_ac_File+set}\" = set"; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  test "$cross_compiling" = yes &&
-  { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
-echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
+  test "$cross_compiling" = yes && test "$ac_env_KERNEL_DIR_set" != set &&
+  { { echo "$as_me:$LINENO: error: KERNEL_DIR must be set when cross compiling" >&5
+echo "$as_me: error: KERNEL_DIR must be set when cross compiling" >&2;}
    { (exit 1); exit 1; }; }
 if test -r "$KERNEL_DIR/include/linux/kernel.h"; then
   eval "$as_ac_File=yes"
@@ -3017,9 +3017,9 @@
 if eval "test \"\${$as_ac_File+set}\" = set"; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  test "$cross_compiling" = yes &&
-  { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
-echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
+  test "$cross_compiling" = yes && test "$ac_env_KERNEL_DIR_set" != set &&
+  { { echo "$as_me:$LINENO: error: KERNEL_DIR must be set when cross compiling" >&5
+echo "$as_me: error: KERNEL_DIR must be set when cross compiling" >&2;}
    { (exit 1); exit 1; }; }
 if test -r "$KERNEL_DIR/kernel/fork.c"; then
   eval "$as_ac_File=yes"
@@ -3064,9 +3064,9 @@
 if eval "test \"\${$as_ac_File+set}\" = set"; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  test "$cross_compiling" = yes &&
-  { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
-echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
+  test "$cross_compiling" = yes && test "$ac_env_KERNEL_DIR_set" != set &&
+  { { echo "$as_me:$LINENO: error: KERNEL_DIR must be set when cross compiling" >&5
+echo "$as_me: error: KERNEL_DIR must be set when cross compiling" >&2;}
    { (exit 1); exit 1; }; }
 if test -r "$KERNEL_DIR/include/linux/version.h"; then
   eval "$as_ac_File=yes"
@@ -3106,9 +3106,9 @@
 if eval "test \"\${$as_ac_File+set}\" = set"; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  test "$cross_compiling" = yes &&
-  { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
-echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
+  test "$cross_compiling" = yes && test "$ac_env_KERNEL_DIR_set" != set &&
+  { { echo "$as_me:$LINENO: error: KERNEL_DIR must be set when cross compiling" >&5
+echo "$as_me: error: KERNEL_DIR must be set when cross compiling" >&2;}
    { (exit 1); exit 1; }; }
 if test -r "$KERNEL_DIR/include/asm/ptrace.h"; then
   eval "$as_ac_File=yes"
@@ -3234,9 +3234,9 @@
 if eval "test \"\${$as_ac_File+set}\" = set"; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  test "$cross_compiling" = yes &&
-  { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
-echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
+  test "$cross_compiling" = yes && test "$ac_env_KERNEL_DIR_set" != set &&
+  { { echo "$as_me:$LINENO: error: KERNEL_DIR must be set when cross compiling" >&5
+echo "$as_me: error: KERNEL_DIR must be set when cross compiling" >&2;}
    { (exit 1); exit 1; }; }
 if test -r "$KERNEL_CHK_FILE"; then
   eval "$as_ac_File=yes"
@@ -5116,9 +5116,9 @@
 if test "${ac_cv_file__etc_lids_lids_conf+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  test "$cross_compiling" = yes &&
-  { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
-echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
+  test "$cross_compiling" = yes && test "$ac_env_KERNEL_DIR_set" != set &&
+  { { echo "$as_me:$LINENO: error: KERNEL_DIR must be set when cross compiling" >&5
+echo "$as_me: error: KERNEL_DIR must be set when cross compiling" >&2;}
    { (exit 1); exit 1; }; }
 if test -r "/etc/lids/lids.conf"; then
   ac_cv_file__etc_lids_lids_conf=yes
@@ -5139,9 +5139,9 @@
 if test "${ac_cv_file__etc_lids_lids_cap+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  test "$cross_compiling" = yes &&
-  { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
-echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
+  test "$cross_compiling" = yes && test "$ac_env_KERNEL_DIR_set" != set &&
+  { { echo "$as_me:$LINENO: error: KERNEL_DIR must be set when cross compiling" >&5
+echo "$as_me: error: KERNEL_DIR must be set when cross compiling" >&2;}
    { (exit 1); exit 1; }; }
 if test -r "/etc/lids/lids.cap"; then
   ac_cv_file__etc_lids_lids_cap=yes
@@ -5162,9 +5162,9 @@
 if test "${ac_cv_file__etc_lids_lids_boot_conf+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  test "$cross_compiling" = yes &&
-  { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
-echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
+  test "$cross_compiling" = yes && test "$ac_env_KERNEL_DIR_set" != set &&
+  { { echo "$as_me:$LINENO: error: KERNEL_DIR must be set when cross compiling" >&5
+echo "$as_me: error: KERNEL_DIR must be set when cross compiling" >&2;}
    { (exit 1); exit 1; }; }
 if test -r "/etc/lids/lids.boot.conf"; then
   ac_cv_file__etc_lids_lids_boot_conf=yes
@@ -5185,9 +5185,9 @@
 if test "${ac_cv_file__etc_lids_lids_boot_cap+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  test "$cross_compiling" = yes &&
-  { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
-echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
+  test "$cross_compiling" = yes && test "$ac_env_KERNEL_DIR_set" != set &&
+  { { echo "$as_me:$LINENO: error: KERNEL_DIR must be set when cross compiling" >&5
+echo "$as_me: error: KERNEL_DIR must be set when cross compiling" >&2;}
    { (exit 1); exit 1; }; }
 if test -r "/etc/lids/lids.boot.cap"; then
   ac_cv_file__etc_lids_lids_boot_cap=yes
@@ -5208,9 +5208,9 @@
 if test "${ac_cv_file__etc_lids_lids_postboot_conf+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  test "$cross_compiling" = yes &&
-  { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
-echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
+  test "$cross_compiling" = yes && test "$ac_env_KERNEL_DIR_set" != set &&
+  { { echo "$as_me:$LINENO: error: KERNEL_DIR must be set when cross compiling" >&5
+echo "$as_me: error: KERNEL_DIR must be set when cross compiling" >&2;}
    { (exit 1); exit 1; }; }
 if test -r "/etc/lids/lids.postboot.conf"; then
   ac_cv_file__etc_lids_lids_postboot_conf=yes
@@ -5231,9 +5231,9 @@
 if test "${ac_cv_file__etc_lids_lids_postboot_cap+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  test "$cross_compiling" = yes &&
-  { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
-echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
+  test "$cross_compiling" = yes && test "$ac_env_KERNEL_DIR_set" != set &&
+  { { echo "$as_me:$LINENO: error: KERNEL_DIR must be set when cross compiling" >&5
+echo "$as_me: error: KERNEL_DIR must be set when cross compiling" >&2;}
    { (exit 1); exit 1; }; }
 if test -r "/etc/lids/lids.postboot.cap"; then
   ac_cv_file__etc_lids_lids_postboot_cap=yes
@@ -5254,9 +5254,9 @@
 if test "${ac_cv_file__etc_lids_lids_shutdown_conf+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  test "$cross_compiling" = yes &&
-  { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
-echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
+  test "$cross_compiling" = yes && test "$ac_env_KERNEL_DIR_set" != set &&
+  { { echo "$as_me:$LINENO: error: KERNEL_DIR must be set when cross compiling" >&5
+echo "$as_me: error: KERNEL_DIR must be set when cross compiling" >&2;}
    { (exit 1); exit 1; }; }
 if test -r "/etc/lids/lids.shutdown.conf"; then
   ac_cv_file__etc_lids_lids_shutdown_conf=yes
@@ -5277,9 +5277,9 @@
 if test "${ac_cv_file__etc_lids_lids_shutdown_cap+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  test "$cross_compiling" = yes &&
-  { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
-echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
+  test "$cross_compiling" = yes && test "$ac_env_KERNEL_DIR_set" != set &&
+  { { echo "$as_me:$LINENO: error: KERNEL_DIR must be set when cross compiling" >&5
+echo "$as_me: error: KERNEL_DIR must be set when cross compiling" >&2;}
    { (exit 1); exit 1; }; }
 if test -r "/etc/lids/lids.shutdown.cap"; then
   ac_cv_file__etc_lids_lids_shutdown_cap=yes
@@ -5300,9 +5300,9 @@
 if test "${ac_cv_file__etc_lids_lids_net+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  test "$cross_compiling" = yes &&
-  { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
-echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
+  test "$cross_compiling" = yes && test "$ac_env_KERNEL_DIR_set" != set &&
+  { { echo "$as_me:$LINENO: error: KERNEL_DIR must be set when cross compiling" >&5
+echo "$as_me: error: KERNEL_DIR must be set when cross compiling" >&2;}
    { (exit 1); exit 1; }; }
 if test -r "/etc/lids/lids.net"; then
   ac_cv_file__etc_lids_lids_net=yes
@@ -5323,9 +5323,9 @@
 if test "${ac_cv_file__etc_lids_lids_ini+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  test "$cross_compiling" = yes &&
-  { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
-echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
+  test "$cross_compiling" = yes && test "$ac_env_KERNEL_DIR_set" != set &&
+  { { echo "$as_me:$LINENO: error: KERNEL_DIR must be set when cross compiling" >&5
+echo "$as_me: error: KERNEL_DIR must be set when cross compiling" >&2;}
    { (exit 1); exit 1; }; }
 if test -r "/etc/lids/lids.ini"; then
   ac_cv_file__etc_lids_lids_ini=yes
@@ -5346,9 +5346,9 @@
 if test "${ac_cv_file__etc_lids_lids_pw+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  test "$cross_compiling" = yes &&
-  { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
-echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
+  test "$cross_compiling" = yes && test "$ac_env_KERNEL_DIR_set" != set &&
+  { { echo "$as_me:$LINENO: error: KERNEL_DIR must be set when cross compiling" >&5
+echo "$as_me: error: KERNEL_DIR must be set when cross compiling" >&2;}
    { (exit 1); exit 1; }; }
 if test -r "/etc/lids/lids.pw"; then
   ac_cv_file__etc_lids_lids_pw=yes
diff -Nur lids-1.2.2-2.4.28/lidstools-0.5.6/src/Makefile.in lids-1.2.2-2.4.28-greenphone/lidstools-0.5.6/src/Makefile.in
--- lids-1.2.2-2.4.28/lidstools-0.5.6/src/Makefile.in	2004-02-17 16:20:27.000000000 +1000
+++ lids-1.2.2-2.4.28-greenphone/lidstools-0.5.6/src/Makefile.in	2007-02-12 15:51:04.000000000 +1000
@@ -38,10 +38,10 @@
 pkgincludedir = $(includedir)/@PACKAGE@
 top_builddir = ..
 
-ACLOCAL = @ACLOCAL@
-AUTOCONF = @AUTOCONF@
-AUTOMAKE = @AUTOMAKE@
-AUTOHEADER = @AUTOHEADER@
+#ACLOCAL = @ACLOCAL@
+#AUTOCONF = @AUTOCONF@
+#AUTOMAKE = @AUTOMAKE@
+#AUTOHEADER = @AUTOHEADER@
 
 am__cd = CDPATH="$${ZSH_VERSION+.}$(PATH_SEPARATOR)" && cd
 INSTALL = @INSTALL@
