--- glibc-2.3.6/csu/Makefile.orig	2006-09-23 07:56:43.000000000 +1000
+++ glibc-2.3.6/csu/Makefile	2006-09-23 07:57:11.000000000 +1000
@@ -241,7 +241,7 @@
 	 esac; \
 	 files="$(all-Banner-files)";				\
 	 if test -n "$$files"; then				\
-	   echo "\"Available extensions:\\n\"";			\
+	   printf '"Available extensions:\\n"\n';			\
 	   sed -e '/^#/d' -e 's/^[[:space:]]*/	/'		\
 	       -e 's/^\(.*\)$$/\"\1\\n\"/' $$files;		\
 	 fi) > $@T
