--- glibc-2.3.6/sysdeps/unix/sysv/linux/arm/socket.S.orig	2006-09-25 08:40:00.000000000 +1000
+++ glibc-2.3.6/sysdeps/unix/sysv/linux/arm/socket.S	2006-09-25 08:41:23.000000000 +1000
@@ -32,7 +32,11 @@
    The .S files for the other calls just #define socket and #include this.  */
 
 #ifndef __socket
-#define __socket P(__,socket)
+# ifndef NO_WEAK_ALIAS
+#  define __socket P(__,socket)
+# else
+#  define __socket socket
+# endif
 #endif
 
 #define PUSHARGS_1	str a1, [sp, $-4]!
@@ -120,4 +124,6 @@
 
 PSEUDO_END (__socket)
 
+#ifndef NO_WEAK_ALIAS
 weak_alias (__socket, socket)
+#endif
