--- ppp-2.4.3/pppd/Makefile.linux	2004-11-13 22:02:22.000000000 +1000
+++ ppp-2.4.3-greenphone/pppd/Makefile.linux	2006-10-23 16:55:54.000000000 +1000
@@ -48,7 +48,7 @@
 # Uncomment the next line to include support for PPP packet filtering.
 # This requires that the libpcap library and headers be installed
 # and that the kernel driver support PPP packet filtering.
-FILTER=y
+#FILTER=y
 
 # Uncomment the next line to enable multilink PPP (enabled by default)
 # Linux distributions: Please leave multilink ENABLED in your builds
