diff -ur bootchart-0.9/script/bootchartd.conf bootchart-0.9_modified/script/bootchartd.conf
--- bootchart-0.9/script/bootchartd.conf	2005-11-14 03:40:01.000000000 +1000
+++ bootchart-0.9_modified/script/bootchartd.conf	2007-08-21 11:21:30.000000000 +1000
@@ -10,7 +10,7 @@
 BOOTLOG_LOCK=".lock"
 
 # Sampling period (in seconds)
-SAMPLE_PERIOD=0.2
+SAMPLE_PERIOD=0.5
 
 # Whether to enable and store BSD process accounting information.  The
 # kernel needs to be configured to enable v3 accounting
