--- gcc-4.1.1/gcc/config/arm/linux-elf.h.orig	2006-09-23 13:00:19.000000000 +1000
+++ gcc-4.1.1/gcc/config/arm/linux-elf.h	2006-09-23 13:00:39.000000000 +1000
@@ -49,7 +49,7 @@
    %{shared:-lc} \
    %{!shared:%{profile:-lc_p}%{!profile:-lc}}"
 
-#define LIBGCC_SPEC "%{msoft-float:-lfloat} %{mfloat-abi=soft*:-lfloat} -lgcc"
+#define LIBGCC_SPEC "-lgcc"
 
 #define LINUX_TARGET_INTERPRETER "/lib/ld-linux.so.2"
 
